File: demodata/android/StringProvider.java
package android;

import java.util.Random;

public class StringProvider {
    public static String getValue(String key, String defaultValue) {
        Random rand = new Random();
        int nextInt = rand.nextInt();
        if (nextInt % 2 == 0) {
            return defaultValue;
        } else {
            return String.format("value %d", nextInt);
        }
    }
}

File: demodata/ios/StringProvider.m
#import "StringProvider.h"

@implementation StringProvider

+ (StringProvider *)sharedInstance {
    static StringProvider *sharedInstance;
    @synchronized (self) {
        if (!sharedInstance) {
            StringProvider *instance =[[StringProvider alloc] init];
            sharedInstance = instance;
        }
        return sharedInstance;
    }
}

- (NSString *)getValue:(NSString *)strKey defaultValue:(NSString *)defaultValue {
    int nextInt = arc4random_uniform(10000);
    if (nextInt % 2 == 0) {
        return defaultValue;
    } else {
        return [NSString stringWithFormat:@"value %d", nextInt];
    }
}

@end
